src/Chapter3.hs

```diff
@@ -907,9 +980,36 @@ Implement instances of "Append" for the following types:
   ✧ *(Challenge): "Maybe" where append is appending of values inside "Just" constructors
 
 -}
+newtype Gold = MkGold Int deriving (Show)
+
+data List a
+    = Empty
+    | Cons a (List a)
+
 class Append a where
     append :: a -> a -> a
+    append a _ = a
 
+instance Append Gold where
+    append :: Gold -> Gold -> Gold
+    append (MkGold a) (MkGold b) = MkGold (a+b)
+
+instance Append (List a) where
+    append :: List a -> List a -> List a
+    append Empty b = b
+    append a Empty = a
+    append (Cons x xs) b = Cons x (append xs b)
+
+-- instance Append (Maybe a) where
+--   append :: (Maybe a) -> (Maybe a) -> (Maybe a)
+--   append Nothing _ = Nothing
+--   append _ Nothing = Nothing
+--   append (Just a) (Just b) = Just (append a b) --this is giving an error
+--this is annoying!! I think I'm close I just don't know how to let Append handle generic types yknow?
```

Comment:
You are almost correct! The body of the `append` function is absolutely correct :gem: 
The problem is when you are trying to do `append a b`. Variables with names  `a` and `b` have type `a`. But to call `append` on values of type `a`, you need to know that `a` is _appendable_. And this can be fixed by requiring `Append` for type `a`. In other words, you need to add additional context (constraint) to the `Append` instance for `Maybe`. And it will compile (after uncommenting the code of course) :slightly_smiling_face: 

```suggestion
-- instance Append a => Append (Maybe a) where
--   append :: (Maybe a) -> (Maybe a) -> (Maybe a)
--   append Nothing _ = Nothing
--   append _ Nothing = Nothing
--   append (Just a) (Just b) = Just (append a b) --this is giving an error
--this is annoying!! I think I'm close I just don't know how to let Append handle generic types yknow?
```

The idea is that you can do `append (Just (Gold 3)) (Just (Gold 4))` because `Gold` implements `Append` instance. But you can't `append (Just True) (Just False)` because `Bool` doesn't have `Append` instance, and it will be a compiler error.